test/tests/functions/datetime.json

```diff
@@ -45,7 +45,7 @@
       "src": "{|2006-01-02T15:04:06| :datetime}"
     },
     {
-      "src": "{|2006-01-02T15:04:06| :datetime year=numeric month=|2-digit|}"
+      "src": "{|2006-01-02T15:04:06| :datetime year=numeric month=2-digit}"
```

Comment:
With this change the parsers now need a pretty big lookahead.

Before you it was enough to look at the first character:
`|` => quoted string
`0`-`9` or '-' > try to get a number literal
starting char => literal
anything else => Error

Reply:
How does that change? Anywhere that literal is valid, single-character lookahead still suffices.
* `|` → quoted-literal
* `$` → variable
* `:` → function
* `*` → key
* `#` or `/` → (markup)
* name-char → unquoted-literal
* anything else → error

Reply:
I agree with @gibson042. 

It is trivial to check the first character. A number literal can only start with one of 11 characters: - or 0..9. All of these are contained in `name-char`:

```
name-char  = name-start / DIGIT / "-" / "."
           / %xB7 / %x300-36F / %x203F-2040
```

So the first character is enough to send you down the path.

Of course, once you start down the path to any of these outcomes, you could end up with something bogus